test/tools/yulInterpreter/EwasmBuiltinInterpreter.cpp

```diff
@@ -133,7 +130,19 @@ u256 EwasmBuiltinInterpreter::evalBuiltin(YulString _fun, vector<u256> const& _a
 		accessMemory(arg[0], 4);
 		return readMemoryHalfWord(arg[0]);
 	}
-
+	else if (_fun == "i32.clz"_yulstring)
+	{
+		if (arg[0] == 0)
```

Comment:
This conditional is here because Wasm specifies determinism with clz/ctz/popcnt, but the instructions themselves are not clear about the 0 input case, at least on x86. Hence why I moved this check here.

Reply:
Aren't they?
The implementation of ``clz`` above should exactly output ``64`` for ``0``, so it shouldn't it be fine without the conditional?

Reply:
Actually no...

Reply:
Why did you remove the ``_v == 0`` case in there? I not sure I understand your reasoning here.

Reply:
Also about specification: https://github.com/WebAssembly/design/blob/master/Semantics.md reads very much like this is fully specified:
``sign-agnostic count leading zero bits (All zero bits are considered leading if the value is zero)``

Reply:
[So from what you're saying in the channel I assume you meant that the compiler builtins, if we chose to use them, are undefined for zero as input - but I'd say I'm not sure that has to dictate that we move the check out of our helper and into here, while keeping our helper either undefined or (in its current form) even dangerous (i.e. spinning) for zero input]

Reply:
I think that @axic wants to be able to use `clz()` and compiler builtins interchangeably here. Since we would need this `if` here if we used a builtin, having it here with `clz()` makes it a drop-in equivalent. And if the check is here, we don't need it inside `clz()`.

I think it makes sense as long as we have some kind of sanity check to prevent the infinite loop situation. I think that a plain `assert()` would be enough.